odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -464,6 +470,47 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.multi
+    def _update_message_from(self, message):
+        """Update the FROM header on message to align with server settings.
+
+        Args:
+            message (email.message): The message object to be evaluated and
+                possibly mutated.
+        """
+
+        self.ensure_one()
+
+        if not self.smtp_from:
+            return
+
+        message.replace_header('Reply-To', message['From'])
```

Comment:
Why replace the "Reply-To" address? Reply-To should have been set properly to ensure that replies come back to Odoo, so it should be preserved, otherwise we'll lose replies, won't we? It should not be relevant for DMARC either.